test/adapters/adapters.spec.js

```diff
@@ -0,0 +1,145 @@
+                                    juttle], { quiet: true });
+}
+
+function waitForSuccess(containerName, args) {
+    return retry(() => {
+        return dockerUtils.execContainer(containerName, args, { quiet: true });
+    }, {
+        max_tries: 10,
+        interval: 1000
+    });
+}
+
+describe('adapters', () => {
+
+    before(() => {
+        var tmpdir = tmp.dirSync().name;
+        return Promise.all([
+            dockerUtils.destroyContainer('elasticsearch-adapter-test'),
+            dockerUtils.destroyContainer('influxdb-adapter-test'),
+            dockerUtils.destroyContainer('juttle-engine-local')
+        ])
+        .then(() => {
+            return dockerUtils.checkJuttleEngineLocalExists();
+        })
+        .then(() => {
+            return Promise.all([
+                dockerUtils.runContainer({
+                    name: 'elasticsearch-adapter-test',
+                    image: 'elasticsearch:1.5.2',
+                    detach: true
+                }),
+                dockerUtils.runContainer({
+                    name: 'influxdb-adapter-test',
+                    image: 'tutum/influxdb:0.10',
+                    detach: true
+                })
+            ]);
+        })
+        .then(() => {
+            var config = JSON.stringify({
+                adapters: {
+                    elastic: {
+                        address: 'elasticsearch',
+                        port: 9200
+                    },
+                    influx: {
+                        url: 'http://influxdb:8086'
+                    }
+                }
+            });
+
+            var filename = path.join(tmpdir, '.juttle-config.json')
+            return fs.writeFileAsync(filename, config)
+        })
+        .then(() => {
+            return dockerUtils.runContainer({
+                name: 'juttle-engine-local',
+                image: 'juttle-engine:local',
+                ports: ['8080:8080'],
+                links: [
+                    'elasticsearch-adapter-test:elasticsearch',
+                    'influxdb-adapter-test:influxdb'
+                ],
+                volumes: [`${tmpdir}:/tmp`],
+                workdir: '/tmp',
+                detach: true
+            });
+        })
+        .then(() => {
+            return waitForSuccess('elasticsearch-adapter-test',
+                                  ['curl', 'http://localhost:9200']);
+        });
+    });
+
+    after(() => {
+        return Promise.all([
+            dockerUtils.destroyContainer('elasticsearch-adapter-test'),
+            dockerUtils.destroyContainer('influxdb-adapter-test'),
+            dockerUtils.destroyContainer('juttle-engine-local')
+        ]);
+    });
+
+    describe('juttle-elastic-adapter', () => {
+        it('can write data and read back data using the juttle CLI', () => {
+            var id = uuid.v1();
+            return juttleBin(`emit -limit 5 -from :2014-01-01: | put name="test-${id}",count=count() | write elastic`)
```

Comment:
Just wanting a fixed data so we can set the expectations below correctly, otherwise we'd have to ignore the `time` field all together.